bluemira/equilibria/find.py

```diff
@@ -725,6 +728,90 @@ def get_flux_loop(psi_norm):
     return lcfs, separatrix
 
 
+def get_legs(equilibrium):
+    """
+    Get the legs of a separatrix.
+
+    Parameters
+    ----------
+    equilibrium: Equilibrium
+        Equilibrium for which to find the separatrix legs
+
+    Returns
+    -------
+    legs: List[Loop]
+        Geometries of the legs, sorted as:
+            [lower inner,  # if present
+             lower outer,  # if present
+             upper inner,  # if present
+             upper outer,  # if present
+            ]
+
+    Raises
+    ------
+    EquilibriaError: if a strange number of legs would be found for an X-point
+
+    Notes
+    -----
+    Will return two legs in the case of a single null
+    Will return four legs in the case of a double null
+    """
+
+    def extract_leg(flux_line, x_point):
+        radial_line = Loop(
+            x=[x_point.x - delta, x_point.x + delta], z=[x_point.z, x_point.z]
+        )
+        arg_inters = join_intersect(flux_line, radial_line, get_arg=True)
+        arg_inters.sort()
+        if x_point.z < o_p.z:
+            # Lower null
+            func = operator.lt
+        else:
+            # Upper null
+            func = operator.gt
+
+        if len(arg_inters) > 2:
+            EquilibriaError(
+                "Unexpected number of intersections with the separatrix around the X-point."
+            )
+
+        flux_legs = []
+        for arg in arg_inters:
+            if func(flux_line.z[arg + 1], flux_line.z[arg]):
+                leg = Loop.from_array(flux_line[arg:])
+            else:
+                leg = Loop.from_array(flux_line[: arg + 1])
+
+            # Make the leg flow away from the plasma core
+            if leg.argmin(x_point[:2]) > 3:
+                leg.reverse()
+
+            flux_legs.append(leg)
+        if len(flux_legs) == 1:
+            flux_legs = flux_legs[0]
+        return flux_legs
+
+    o_points, x_points = equilibrium.get_OX_points()
+    o_p = o_points[0]
```

Comment:
Would making `o_p` an argument in `extract_leg` work? It took some digging to find where this definition was being used.

Reply:
it's the same O-point every time, I will rename o_p and x_p though to make it clearer. Sonarcloud thinks this is too complicated anyway, so I will try and refactor a bit.